### PR TITLE
x-create-mouse-void: init at 0.1

### PR DESCRIPTION
**Pull request by eigengrau**
###### Description of changes

X-Create-Mouse-Void creates an undecorated black window and prevents the mouse from entering that window. This is useful in multi-monitor configurations of Xorg when the monitors have different geometries (e.g., one screen vertical, one horizontal), since it allows constraining the mouse movement to visible parts of the Xorg screen.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14589,4 +14589,10 @@
     github = "npatsakula";
     githubId = 23001619;
   };
+  eigengrau = {
+    email = "seb@schattenkopie.de";
+    name = "Sebastian Reuße";
+    github = "eigengrau";
+    githubId = 4939947;
+  };
```
Comment (juboba) [inline maintainers/maintainer-list.nix:14597]: This should be in a separate commit.

**Review by juboba — CHANGES_REQUESTED** [review]
Hi @eigengrau!

Just a minor change request.

Thank you!

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (eigengrau) [reply maintainers/maintainer-list.nix:14597]: Hi! Could you clarify? This change is in 9712edc984d, which is already separate from the one adding the package (4d84afde092).

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, xorg, fetchFromGitHub }:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/x-create-mouse-void/default.nix:2]: Missing new line

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "0.1";
+    sha256 = "151pv4gmzz9g6nd1xw94hmawlb5z8rgs1jb3x1zpvn3znd7f355c";
+  };
+
+  buildInputs = [ xorg.libX11 ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -pv $out/bin
+    cp -av XCreateMouseVoid $out/bin/x-create-mouse-void
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/x-create-mouse-void/default.nix:18]: Remove the -v

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/cas--/XCreateMouseVoid";
+    description =
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/x-create-mouse-void/default.nix:24]: Here shouldn't be a line break.

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      "Creates an undecorated black window and prevents the mouse from entering that window";
+    platforms = platforms.unix;
+    license = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/x-create-mouse-void/default.nix:27]: The package does not really have a license so we must set license.unfree.

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, xorg, fetchFromGitHub }:
+stdenv.mkDerivation rec {
+  pname = "x-create-mouse-void";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "cas--";
+    repo = "XCreateMouseVoid";
+    rev = "0.1";
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/x-create-mouse-void/default.nix:9]: This should be taken from version.

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix (hunk 3 above)
Comment (eigengrau) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: Applied. (To save storage in the caches, I guess?)

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix (hunk 4 above)
Comment (eigengrau) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: Applied (still had nixfmt configured in Emacs).

File: pkgs/applications/window-managers/x-create-mouse-void/default.nix (hunk 5 above)
Comment (eigengrau) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: Applied. 

There is a license granted, but I’m not sure what our definition of “free” is. Any license explicitly listed by FSF/GNU as libre?

Do you think “unfree-redistributable” would also work?

Comment (SuperSandro2000) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: free licenses must at least allow us to freely distribute the source code and compilation.



> Do you think “unfree-redistributable” would also work?

I think that would work, too.

Comment (eigengrau) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: Thanks! Amended…

Comment (06kellyjac) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: https://github.com/cas--/XCreateMouseVoid/blob/master/XCreateMouseVoid.c#L1-L6

unfree redistributable will do for now. The comment suggests they're pretty flexible but you can always email patrick & cas-- and maybe get them to explicitly agree to something like GPLv3 or WTFPL :shrug:

Comment (06kellyjac) [reply pkgs/applications/window-managers/x-create-mouse-void/default.nix]: also welcome to nixpkgs @eigengrau 

I noticed you maintain this package on the AUR, is this a sign you'll maintain more stuff in nixpkgs going forward? :rocket:
